src/components/Sender/Sender.tsx

```diff
@@ -0,0 +1,62 @@
+import { Button, Grid, TextField } from '@material-ui/core';
+import React from 'react';
+import { MailTemplateCtx } from '../../contexts/mail-template.context';
+import { SpreadsheetCtx } from '../../contexts/spreadsheet.context';
+import { UserCtx } from '../../contexts/user.context';
+import send from '../../services/MailSender';
+import { replaceVars } from '../utils';
+import Recipients from './Recipients';
+
+const Sender = () => {
+  const [mailTemplate] = React.useContext(MailTemplateCtx);
+  const [spreadsheet] = React.useContext(SpreadsheetCtx);
+  const [user] = React.useContext(UserCtx);
+  const [subject, setSubject] = React.useState('Proszę o wystawienie Faktury');
+
+  const recipients = spreadsheet.usersData.filter((user: any) => user.send);
+  const dataToSend = recipients.map((userData: any) => {
+    if (userData.send) {
+      return {
+        email: userData.email,
+        data: {
+          subject,
+          content: replaceVars(mailTemplate, userData),
+        },
+      };
+    }
+    return;
+  });
+
+  const SendEmails = () => {
+    send(dataToSend, user);
+  };
+
+  const changeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSubject(event.target.value);
+  };
+
+  return (
+    <Grid item={true} xs={12} style={{ textAlign: 'center' }}>
+      <TextField
+        id="standard-search"
+        label="Email Title"
+        type="search"
+        margin="normal"
+        style={{ width: '90%', margin: 'auto' }}
+        value={subject}
+        onChange={changeTitle}
+      />
+      <Recipients />
+      <Button
+        variant="contained"
+        color="primary"
+        onClick={SendEmails}
+        disabled={!recipients.length}
+      >
+        Send
+      </Button>
+    </Grid>
+  );
+};
+
+export default Sender;
```

Comment:
Wrap it with `memo` to make sure component won't re-render if props didn't change

Reply:
Can you explain what `memo` exactly is? I just quickly looked at the docs but it doesn't tell much

Reply:
`memo` function creates memoized component. It will be re-rendered only when its props change. If your component is a child of other component, and parent must re-render, your component will also re-render if it's not memoized even when props didn't change. This is why you want to wrap your function components with `memo`. Read more here https://scotch.io/tutorials/react-166-reactmemo-for-functional-components-rendering-control#toc-what-is-react-memo-

Reply:
that's exhaustive answer, I like it 😎